File: 6_kyu/Are_they_the_same.lua
local solution = {}

-- Checks if the table b contains all number of table a squared.
-- The order of the items could be different.
function solution.comp(a, b)
  if type(a) ~= 'table' or type(b) ~= 'table' or #a ~= #b then
    return false
  end
  table.sort(a, function(x1,x2) return math.abs(x1)<math.abs(x2) end)
  table.sort(b)
  for i = 1,#a do
    if a[i]^2 ~= b[i] then
      return false
    end
  end
  return true
end

return solution

File: 6_kyu/Are_they_the_same.r
# Checks if the vector a2 contains all number of vector a1 squared.
# The order of the items could be different.
# The values are all integers, compare is done with absolute limit 0.01.
comp <- function(a1, a2) {
  if( length(a1) != length(a2) )
    return(FALSE)
  # it is a**2 = (-a)**2, for fast compare use only the part 
  a <- sort(abs(a1))
  b <- sort(a2)
  # Values should be all integer values.
  # To avoid floating point rounding errors, use the 0.01 limit.
  !any( abs(a**2 - b) > 0.01 )
}
